quam_builder/architecture/quantum_dots/virtual_gates/README.md

```diff
@@ -0,0 +1,231 @@
+
+Virtual gates simplify complex tuning procedures in experiments, especially for spin qubits, by allowing control over abstract parameters that map to multiple physical gate voltages.
+
+**`VirtualGateSet` builds upon `GateSet` and inherits all of its features**, including physical channel management, `VoltageTuningPoint` definitions, and all voltage control methods. This means you can use virtual gates while still having access to all the voltage control capabilities of the underlying `GateSet`.
+
+## 2. Core Components
+
+### 2.1 VirtualGateSet
+
+A subclass of `GateSet`. It manages a list of `VirtualisationLayer` objects, which define the transformations from virtual gate voltages to underlying (either physical or lower-level virtual) gate voltages.
+
+### 2.2 VirtualisationLayer
+
+Represents a single linear transformation (matrix) from a set of source (virtual) gates to a set of target gates.
+
+## 3. VirtualGateSet
+
+A `VirtualGateSet` allows users to define and operate with virtual gates, abstracting the underlying physical gate operations.
+
+**Key Features:**
+
+- **Inherits from `GateSet`:** Retains all functionalities of `GateSet`, including physical channel management and `VoltageTuningPoint` definitions.
+- **Manage Multiple Virtualisation Layers:** Stores a list of `VirtualisationLayer` objects. Multiple layers can be defined and stacked, allowing for hierarchical virtualisation. Layers are applied sequentially (in reverse order during voltage resolution) to translate top-level virtual gate voltages into physical gate voltages.
+- **Add Layers:** Use `add_layer(source_gates, target_gates, matrix)` to define and append a new `VirtualisationLayer`:
+  - `source_gates`: Names of the new virtual gates defined by this layer.
+  - `target_gates`: Names of the gates (physical or virtual from a previous layer) that this layer maps onto.
+  - `matrix`: The transformation matrix (list of lists of floats).
+- **Additive Voltage Resolution:** Overrides `GateSet.resolve_voltages()`. When voltages are specified for virtual gates (potentially across different layers) and/or physical gates simultaneously, this method applies the inverse of the virtualisation matrices for each layer. Contributions from all specified virtual and physical gates are resolved and become additive at the physical gate level. Handles multi-layered virtualisation by processing layers from the outermost to the innermost.
+
+### 3.1 Defining a `VirtualGateSet`
+
+```python
+from quam.components import SingleChannel
+from quam_builder.architecture.quantum_dots.virtual_gates import VirtualGateSet
+
+# Assume channel_P1, channel_P2 are existing SingleChannel objects
+physical_channels = {"P1": channel_P1, "P2": channel_P2}
+v_gate_set = VirtualGateSet(id="virt_dot_gates", channels=physical_channels)
```

Comment:
I am a bit confused, should the `VirtualGateSet `be initialized with a `GateSet `object?
Isn't it the same to have a `VirtualGateSet `with no virtualization layer and a simple `GateSet`? If so, why not just having a `GateSet` components with `VirtualizationLayers`?